src/ResizeWatcher.ts

```diff
@@ -0,0 +1,122 @@
+import Component from "@egjs/component";
+import { SizeRect } from "./types";
+import { isString } from "./utils";
+
+export interface ResizeWatherOptions {
+  resizeDebounce?: number;
+  maxResizeDebounce?: number;
+  useResizeObserver?: boolean;
+  useWindowResize?: boolean;
+  watchDirection?: "width" | "height" | "box" | false;
+  rectBox?: "border-box" | "content-box";
+}
+
+export interface OnResize {
+
+}
+
+
+export class ResizeWatcher {
+  private _resizeTimer = 0;
+  private _maxResizeDebounceTimer = 0;
+  private _emitter: Component<{ resize: OnResize }>;
+  private _observer: ResizeObserver | null;
+  protected container: HTMLElement;
+  protected rect: SizeRect = { width: 0, height: 0 };
+  protected options!: Required<ResizeWatherOptions>;
```

Comment:
Why these properties are marked as `protected`? Is there any class that inherits this class?